### PR TITLE
Plant nutrition rebalance: phase one

### PR DESCRIPTION
**Pull request by Saint-of-Grey**
#### Summary
```SUMMARY: Balance: "Update plant mutant nutrition gains to use new nutrition systems"```
#### Purpose of change
This partially addresses #29241, but doesn't fix it completely.

The changes in this PR are dependent on #29907 being merged.

A lot of the plant mutations have been left behind as the numerous nutrition systems have been updated. A quick playthough as a plant shows they will not extract any vitamins, but will draw an obtuse amount of calories and water from the soil.
After some research, I've devised a way to make plant nutrition gains more reflective of reality (well, as close to reality as it can be).
#### Describe the solution
Scattered research has shown me plants get their supply of calcium and iron from the soil, as such, Roots and Rooter now increase a special counter that grants these vitamins upon being filled. There is a similar mechanism used with water.

Leaves has also received some changes, it now also uses counters for kcal, vitA, and vitC (which plants synthesize when left in sunlight). Players now must not be wearing a hat to benefit from leaves.

There are two more levels of leaves that grant even further nutrition, called "Lush Leaves" and "Verdant Leaves". These give more kcal and vitamins, but require player arms to be uncovered. They also add covered encumbrance to arms, similar how roots adds covered encumbrance to feet. Verdant leaves can allow a player to gain their fully daily kcal and vitamin A and C needs, provided they stay in the sunlight for 12 hours a day. The higher two leaves mutations also provide fluctuating output based on ambient temperature.

There is a new mutation called "Transpiration", which causes player thirst rate to increase or decrease based on ambient temperature. It is required in order to mutate the new leaves mutations.

Finally, the "Very heat dependent" and "Ectothermic" mutations have been added to the plant category, both to reflect how plants react to temperature, and to bring calorie consumption to levels that a plant could feasibly produce over 12 hours of sunlight in the new england area.

Plants still do not have a way of producing vitamin B12, other than eating. This will be dealt with in a future PR.
#### Describe alternatives you've considered
This is a slightly more reigned in version of my previous PR, without rooting keeping you stuck in place.

#### Additional context
The next plant-based PR will be focused on giving players with the "Saprophage" trait the ability to root in corpses to drain them of nutrition. This will be how plants can attain vitamin B, but will likely be a source of nutrition for any active plant.

### CODE REVIEW

**Comment by Yunatan** [comment]
I once researched if plants hypothetically can drink blood and apparently it is too thick for them and they have to rely on symbiotic organisms to decompose it first.

**Comment by Saint-of-Grey** [comment]
We don't need to simulate everything within the body, once the plant sucks up the nutrition I can just divert it to the gut before it's available for use. I'm sure plant mutants would have some unique gut fauna to help out with all this.

File: src/player.cpp [hunk 1]
```diff
@@ -7690,21 +7719,18 @@ void player::rooted_message() const
...
-                mod_hunger( -1 );
-                // absorbs nutrients from soil directly
-                mod_stored_nutr( -1 );
-            }
-            if( get_thirst() > -20 ) {
-                mod_thirst( -1 );
-            }
-            mod_healthy_mod( 5, 50 );
+        vitamin_mod( vitamin_id( "counter_roots_vitamins" ), 1, false );
+        if (get_thirst() <= -2000) {
```
Comment (KorGgenT) [inline src/player.cpp]: need spaces on the inside of these parenthesis

File: src/player.h [hunk 2]
```diff
@@ -1853,8 +1855,8 @@ class player : public Character
          * Warmth from terrain, furniture, vehicle furniture and traps.
          * Can be negative.
          **/
-        static int floor_bedding_warmth( const tripoint &pos );
-        /** Warmth from clothing on the floor **/
+        static int floor_bedding_warmth( const tripoint
+                                         &pos );        /** Warmth from clothing on the floor **/
```
Comment (KorGgenT) [inline src/player.h]: looks like a typo caused this comment to be on the same line here. you should change this back to how it was.

Comment (Saint-of-Grey) [reply src/player.h]: ...I'm not even sure what caused that one.

Comment (Saint-of-Grey) [reply src/player.h]: Upon further investigation I found out that astyle really doesn't like how long that line is.

File: src/player.cpp [hunk 3]
```diff
@@ -3202,8 +3205,8 @@ void player::search_surroundings()
         }
         // Chance to detect traps we haven't yet seen.
         if( tr.detect_trap( tp, *this ) ) {
-            if( tr.get_visibility() > 0 ) {
-                // Only bug player about traps that aren't trivial to spot.
+            if( tr.get_visibility() >
+                0 ) {                // Only bug player about traps that aren't trivial to spot.
```
Comment (KorGgenT) [inline src/player.cpp]: looks like some more rogue whitespace here.

File: src/player.cpp [hunk 4]
```diff
@@ -5967,10 +5967,39 @@ void player::suffer()
...
+
+    if( vitamin_get( vitamin_id( "counter_sunlight_vitamin" ) ) >= 3000 ) {
+        vitamin_mod( vitamin_id( "vitA" ), 1, true );
+        vitamin_mod( vitamin_id( "vitC" ), 1, true );
+        vitamin_set( vitamin_id( "counter_sunlight_vitamin" ), 0 );
+    }
+
+    if( vitamin_get( vitamin_id( "counter_sunlight_kcal" ) ) >= 2000 ) {
         mod_hunger( -1 );
-        // photosynthesis absorbs kcal directly
```
Comment (KorGgenT) [inline src/player.cpp]: why did you remove this comment, if you did not remove the line underneath it?

File: src/player.cpp (hunk 3 above)
Comment (Saint-of-Grey) [reply src/player.cpp]: Gah, astyle must really hate me. Anyone else ever get this?

File: src/player.cpp (hunk 4 above)
Comment (Saint-of-Grey) [reply src/player.cpp]: I actually re-wrote that whole bit of code, I didn't really think about the comment. Should I re-add it?

Comment (KorGgenT) [reply src/player.cpp]: the comment was there for the `mod_stored_nutr()` line, which it does not look like you removed. as such, i don't think removing it was necessary. of course, it may be a superfluous comment, but that's probably nitpicking.

Comment (Saint-of-Grey) [reply src/player.cpp]: I throw it back in anyway, for the sake of having some documentation.

**Comment by Saint-of-Grey** [comment]
So it looks like the upcoming change to make turns 1 second instead of 6 is going to mess up my counters. Not sure if I should commit the fix now or later.

**Comment by Yunatan** [comment]
If you believe they are going to get merged into master eventually, you can merge them into you branch and put "depends on X" into this PR description

**Comment by kevingranade** [comment]
Obsoleted by #30884
